cmd/kube-apiserver/app/options/validation.go

```diff
@@ -35,7 +36,10 @@ import (
 func validateClusterIPFlags(options *ServerRunOptions) []error {
 	var errs []error
 	// maxCIDRBits is used to define the maximum CIDR size for the cluster ip(s)
-	const maxCIDRBits = 20
+	maxCIDRBits := 20
+	if utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRServiceAllocator) {
+		maxCIDRBits = 64
```

Comment:
How are we planning a revert? what if somebody started using the feature and then decided nope i want to go back to normal?

Reply:
yeah, reality is that the Services misconfiguration is broken with and without this feature too https://github.com/kubernetes/kubernetes/issues/114743

This will require manual intervention, use a service-cidr with a valid range, and delete manually the Services that will be OutOfRange. The repair loops will recover the old allocators and raise warnings and events for the Services with clusterIPs out of range.

The funny thing is that they may work, at least with kube-proxy, since kube-proxy doesn't have any notion of the service-cidr, it will install the correct iptables rules.

Reply:
ack - please make sure it is part of the documentation.